pkg/kube/inject/inject.go

```diff
@@ -737,8 +738,12 @@ func IntoObject(injector Injector, sidecarTemplate Templates, valuesConfig strin
 			return nil, merr
 		}
 	}
+	var neverInjectSelector []metav1.LabelSelector
+	if sidecarConfig != nil {
+		neverInjectSelector = sidecarConfig.NeverInjectSelector
+	}
 	if patchBytes == nil {
-		if !injectRequired(IgnoredNamespaces, &Config{Policy: InjectionPolicyEnabled}, &pod.Spec, pod.ObjectMeta) {
+		if !injectRequired(IgnoredNamespaces, &Config{Policy: InjectionPolicyEnabled, NeverInjectSelector: neverInjectSelector}, &pod.Spec, pod.ObjectMeta) {
```

Comment:
I am not sure why we are changing the main code when the PR is adding tests. Is this refactoring to make it more testable or fixing a bug?

It would be preferred to only touch test code.

Reply:
@howardjohn - we need to make a change to make it more testable (let me update PR description). In my test I am supplying `NeverInjectSelector` from the operator spec but because this specific code path is executed from the test, where we create a new Config object with just  `InjectionPolicyEnabled` property set on it, its just getting ignored.

Having said that - Since we are supplying `Config` to `injectRequired` , I think it might be cleaner to pass the `Config` object as is instead of specifically passing `Config.Templates` from `IntoResources` and `IntoObject`. WDYT ?